### PR TITLE
gnomeExtensions.dash-to-panel: fix wrongly shown version number and remove manual packaging (closes #157005)

### LINKED ISSUE #157005: Dash to panel gnome shell extension has broken version number
### Describe the bug
Screenshot from 2022-01-27 14-34-18

### Steps To Reproduce
1. nix-env -i gnome-shell-extension-dash-to-panel
2. Open a gnome extension manager and look at the built-in version of the dash to panel extension

### Expected behavior
The version should be 45, not 9999.

### Notify maintainers

https://github.com/mounium

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.93, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(nacia): `""`
 - channels(root): `"nixos-21.11.335481.604c44137d9"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by rhoriguchi**
###### Motivation for this change

The extension sets the version to 9999 in [metadata.json](https://github.com/home-sweet-gnome/dash-to-panel/blob/v45/metadata.json#L9) which then gets shown wrongly to the user. I guess this get's set when the extension get's released.

Closes #157005

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @RDKRACZ

### CODE REVIEW

**Comment by piegamesde** [comment]
Wow, this is awesome. I didn't expect that the automatically packaged extension would work without any modifications.

I think it is safe to remove @Mounium as maintainer.

**Comment by rhoriguchi** [comment]
> Wow, this is awesome. I didn't expect that the automatically packaged extension would work without any modifications.
> 
> I think it is safe to remove @Mounium as maintainer.

I did not test it but I don't see why it would not work. I'll test it later today and mark it till I'm done as a draft. Also, the README.md does not state any requirements and the package has no patch.

**Comment by piegamesde** [comment]
> @piegamesde how do I test this without setting my branch as package source? Usually, I just create an overlay where I pull my branch and build the packaged with super.callPackage.

I don't understand what exactly you are trying to achieve and how. I usually test packages either in a VM with the PR checked out or with using an overlay that overrides that package.

Alternatively, you can *patch* you `nixpkgs` with the pull request diff (it kind of requires an ugly double import of nixpkgs though lol).

**Comment by piegamesde** [comment]
I'd be inclined to merge this now, would you please mark the PR as ready?

**Comment by rhoriguchi** [comment]
> I'd be inclined to merge this now, would you please mark the PR as ready?

I haven't had the time to test it, since I don't use this extension. When I skimmed the extension I just noticed that the build number is being set when released and I didn't see anything that could cause issues.

File: pkgs/desktops/gnome/extensions/dash-to-panel/default.nix [hunk 1]
```diff
@@ -1,31 +0,0 @@
...
-    repo = "dash-to-panel";
-    rev = "v${version}";
-    sha256 = "05bfd3b1g9zd86pl1rpgfqsmip271lasyfj8phpqf1gdds5yz6f6";
-  };
-
-  buildInputs = [
-    glib gettext
-  ];
-
-  makeFlags = [ "INSTALLBASE=$(out)/share/gnome-shell/extensions" ];
```
Comment (jtojnar) [inline pkgs/desktops/gnome/extensions/dash-to-panel/default.nix:18]: Looking at the `Makefile`, you could just pass `"VERSION=${version}"` to `makeFlags`:

https://github.com/home-sweet-gnome/dash-to-panel/blob/940215555b67bd79d9e87f79b4961bd94cf31a45/Makefile#L94-L95

Comment (rhoriguchi) [reply pkgs/desktops/gnome/extensions/dash-to-panel/default.nix:18]: Thank you for the info. I mistaken the gnome extension name from your PR #160726 🤦🏻‍♂️. Getting it directly from <https://extensions.gnome.org> fixes that issue.

**Pull request merged**